apps/services/endorsements/api/src/app/modules/endorsement/endorsement.service.ts

```diff
@@ -68,6 +80,33 @@ export class EndorsementService {
       orderOption: [['counter', 'DESC']],
       where: { endorsementListId: listId },
     })
+
+    // endorsement data display rules for everyone(public,users,admins) except owner ...
+    // ... list owner sees all data
+    const user = this.request.auth as User
+    const listOwnerNationalId = await this.getListOwnerNationalId(listId)
+    if (user?.nationalId != listOwnerNationalId) {
+      endorsements.data.map((endorsement) => {
+        if (!endorsement.meta.showName) {
+          endorsement.meta.fullName = ''
+          endorsement.meta.locality = ''
+        }
+      })
+    }
+    return endorsements
+  }
+
+  async getListOwnerNationalId(listId: string): Promise<string | null> {
```

Comment:
Would it be possible to user the getListOwner function in the endorsementList service that already exists}